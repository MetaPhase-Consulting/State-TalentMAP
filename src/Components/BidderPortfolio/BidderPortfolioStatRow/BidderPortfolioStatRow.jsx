import React from 'react';
import { get } from 'lodash';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { checkFlag } from 'flags';
import { BIDDER_OBJECT, CLASSIFICATIONS } from '../../../Constants/PropTypes';
import SkillCodeList from '../../SkillCodeList';
import { NO_GRADE, NO_POST } from '../../../Constants/SystemMessages';
import ClientBadgeList from '../ClientBadgeList';
import CheckboxList from '../CheckboxList';
import SearchAsClientButton from '../SearchAsClientButton';

const useCDOBidding = () => checkFlag('flags.cdo_bidding');

const BidderPortfolioStatRow = ({ userProfile, showEdit, classifications }) => {
  const currentAssignmentText = get(userProfile, 'pos_location');
  const clientClassifications = get(userProfile, 'classifications');
  return (
    <div className="usa-grid-full bidder-portfolio-stat-row">
      <div className="stat-card-data-point stat-card-data-point--name">
        {get(userProfile, 'name', 'N/A')}
        <Link to={`/profile/public/${userProfile.perdet_seq_number}`}>View Profile</Link>
      </div>
      <div>
        <div>
          <div className="stat-card-data-point">
            <dt>Skill:</dt><dd><SkillCodeList skillCodes={userProfile.skills} /></dd>
          </div>
          <div className="stat-card-data-point">
            <dt>Grade:</dt><dd>{userProfile.grade || NO_GRADE}</dd>
          </div>
          <div className="stat-card-data-point">
            <dt>Location:</dt><dd>{currentAssignmentText || NO_POST}</dd>
          </div>
        </div>
        {
          !showEdit &&
          <div className="bidder-portfolio-stat-row-updates">
            <h4>Classifications: </h4>
            <ClientBadgeList
              clientClassifications={clientClassifications}
              classifications={classifications}
            />
          </div>
        }
        {
          !showEdit && useCDOBidding() &&
          <div className="button-container">
            <SearchAsClientButton user={userProfile} />
          </div>
        }
        {
          showEdit &&
          <CheckboxList id={userProfile.id} />
        }
      </div>
    </div>
  );
};

BidderPortfolioStatRow.propTypes = {
  userProfile: BIDDER_OBJECT.isRequired,
  showEdit: PropTypes.bool,
  classifications: CLASSIFICATIONS,
};

BidderPortfolioStatRow.defaultProps = {
  showEdit: false,
  classifications: [],
};

export default BidderPortfolioStatRow;
