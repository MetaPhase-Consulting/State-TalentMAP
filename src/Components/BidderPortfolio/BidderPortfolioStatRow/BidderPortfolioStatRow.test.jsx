import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import toJSON from 'enzyme-to-json';
import BidderPortfolioStatRow from './BidderPortfolioStatRow';
import { clientObject } from '../../../__mocks__/client';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('BidderPortfolioStatRow', () => {
  const props = {
    userProfile: clientObject,
  };
  it('is defined', () => {
    const wrapper = shallow(
    <Provider store={mockStore({})}>
      <BidderPortfolioStatRow
        {...props}
      />
    </Provider>,);
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <Provider store={mockStore({})}>
        <BidderPortfolioStatRow
          {...props}
        />
      </Provider>,);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
