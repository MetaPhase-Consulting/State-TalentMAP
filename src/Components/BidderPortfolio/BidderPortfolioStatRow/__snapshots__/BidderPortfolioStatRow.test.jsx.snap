// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BidderPortfolioStatRow matches snapshot 1`] = `
<div
  className="usa-grid-full bidder-portfolio-stat-row"
>
  <div>
    <Avatar
      className=""
      fallback={
        <FontAwesome
          name="user-o"
        />
      }
      firstName=""
      initials=""
      lastName=""
      onClick={[Function]}
      small={false}
    />
  </div>
  <div>
    <div
      className="stat-card-data-point stat-card-data-point--name"
    >
      N/A
      <Link
        replace={false}
        to="/profile/public/undefined"
      >
        View Profile
      </Link>
    </div>
    <div
      className="stat-card-data-point"
    >
      <dt>
        Skill:
      </dt>
      <dd>
        <SkillCodeList
          skillCodes={
            Array [
              Object {
                "code": "6218",
                "cone": null,
                "description": "CONSTRUCTION ENGINEERING",
                "id": 63,
              },
            ]
          }
        />
      </dd>
    </div>
    <div
      className="stat-card-data-point"
    >
      <dt>
        Grade:
      </dt>
      <dd>
        01
      </dd>
    </div>
    <div
      className="stat-card-data-point"
    >
      <dt>
        Location:
      </dt>
      <dd>
        None listed
      </dd>
    </div>
  </div>
  <div
    className="bidder-portfolio-stat-row-updates"
  >
    <Connect(StaticDevContent)>
      <ClientBadgeList
        statuses={
          Object {
            "fairshare": 1,
            "handshake": 1,
            "retirement": 2,
            "sixeight": 0,
          }
        }
      />
    </Connect(StaticDevContent)>
  </div>
</div>
`;
